### PR TITLE
PLAT-102263: Migrate SpotlightContainerDecorator to use hooks

### PR DESCRIPTION
**Pull request by ybsung**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

Migrate spotlight/SpotlightContainerDecorator to use hooks

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

Add `SpotlightContainer` for spotlight container behavior.
Add `useSpotlightContainer` hook to manage the life cycle of the SpotlightContainer behavior.
Adopt `spotlight/SpotlightContainerDecorator` to use `useSpotlightContainer`.
Add `useSpot-specs.js`.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

- Moved `releaseContainer` and `stateFromProps` in useSpotlightContainer
- Added the missed JS Doc for `navigableFilter` in spotlight/SpotlightContainerDecorator.SpotlightContainerDecorator.defaultConfig
- Define `ForwardSpotlightContainerDecorator` to escape the following error. The `React.forwardRef` couldn't be used with `propTypes` and `defaultProps`.
```
This will give me the following warning in the console: Warning: forwardRef render functions do not support propTypes or defaultProps. Did you accidentally pass a React component?
```

### Links
[//]: # (Related issues, references)

PLAT-102263

### CODE REVIEW

File: packages/spotlight/SpotlightContainerDecorator/SpotlightContainer.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+				this.releaseContainer(this.id);
+			}
+
+			this.id = Spotlight.add({
+				...this.config,
+				id,
+				restrict,
+				navigableFilter: this.navigableFilter
+			});
+			this.preserveId = this.canPreserveId && this.id === id;
```
Comment (webOS101) [inline packages/spotlight/SpotlightContainerDecorator/SpotlightContainer.js:56]: Does this imply `preserveId` only works if you specify an id?  I think this behaves differently from the previous implementation insofar as a generated id would be passed to `stateFromProps`.  In this case, we should test if `id` is `null` and allow that to preserve the id.

File: packages/spotlight/SpotlightContainerDecorator/useSpotlightContainer.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * Configuration for `useSpotlightContainer`
+ *
+ * @typedef {Object} useSpotlightContainerConfig
+ * @memberof spotlight/SpotlightContainerDecorator
+ * @property {Object}   [containerConfig]                   Spotlight container configuration.
+ * @property {Function} [navigableFilter]                   Called to determine if a spottable
+ *                                                          component is accessible.
+ * @property {Boolean}  [preserveId]                        Preserves the spotlight id when the
```
Comment (webOS101) [inline packages/spotlight/SpotlightContainerDecorator/useSpotlightContainer.js:14]: is it important to call out which properties can be updated and which are only set once at creation?

**Comment by ryanjduffy** [comment]
Pushed doc updates and updated the hook to allow omitting the config object (along with a unit test for that).

I wonder if we should revisit our arg decision to separate design-time and run-time configurations into separate args. I didn't address that here but we can revisit later.

File: packages/spotlight/SpotlightContainerDecorator/tests/useSpotlightContainer-specs.js [hunk 3]
```diff
@@ -48,6 +48,25 @@ describe('useSpotlightContainer', () => {
 	// remove all containers after each test
 	afterEach(Spotlight.clear);
 
+	test('should support omitting the config object', () => {
```
Comment (webOS101) [inline packages/spotlight/SpotlightContainerDecorator/tests/useSpotlightContainer-specs.js:51]: :nice:

**Pull request merged**